rust/src/tensor.rs

```diff
@@ -0,0 +1,381 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Arrow Tensor Type
+use std::marker::PhantomData;
+use std::mem;
+
+use buffer::Buffer;
+use datatypes::{ArrowPrimitiveType, DataType};
+
+/// Computes the strides required assuming a row major memory layout
+fn compute_row_major_strides<T>(shape: &Vec<i64>) -> Vec<i64>
+where
+    T: ArrowPrimitiveType,
+{
+    let mut remaining_bytes = mem::size_of::<T>() as i64;
+    for i in shape {
+        remaining_bytes *= i;
```

Comment:
here you should probably check for overflows using [`checked_mul`](https://doc.rust-lang.org/std/primitive.usize.html#method.checked_mul)